src/android/interoperability/cpp/cpp-bridge.md

```diff
@@ -0,0 +1,52 @@
+# C++ Bridge Declarations
+
+```rust,ignore
+#[cxx::bridge]
+mod ffi {
+{{#include ../../../../third_party/cxx/blobstore/src/main.rs:cpp_bridge}}
+}
+```
+
+Results in (roughly) the following Rust:
+
+```rust,ignore
+#[repr(C)]
```

Comment:
Very nice! During the class, I would want to generate this code and show the students where it is using [`bat`](https://github.com/sharkdp/bat) or similar...

How do I do this?

Reply:
Probably the easiest way to view the generated code is to follow the speaker notes from the Bridge Module slide, i.e.:

* To view the generated Rust code, use [cargo-expand] to view the expanded proc
  macro. For most of the examples you would use `cargo expand ::ffi` to expand
  just the `ffi` module (though this doesn't apply for Android projects).
* To view the generated C++ code, look in `target/cxxbridge`.

You won't easily be able to do this directly with the code snippet in the slide, but you could pull up the blobstore example and build with Cargo to show this. The blobstore example is setup to build with both Cargo and Soong, so the easiest way to demo stuff will probably be with the Cargo version of the project.

[cargo-expand]: https://github.com/dtolnay/cargo-expand